src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
+  const isGroupVisible = (group: ContactFilterGroup) =>
+    group.alwaysVisible || getSelectedFilters(group).length > 0;
+
   return (
-    <div style={{ backgroundColor: colors.amber[600] }}>
-      <h2>Filters</h2>
-      <button onClick={() => loadContactFilters()}>Load Filters</button>
-      {error && <p data-testID="ErrorText">Error: {error.toString()}</p>}
-      {loading ? (
-        <p>Loading Filters</p>
-      ) : !data || data.contactFilters.length === 0 ? (
-        <p>No Filters</p>
-      ) : (
-        <ul data-testID="FiltersList">
-          {data.contactFilters.map(({ id, name }) => (
-            <li key={id}>{name}</li>
-          ))}
-        </ul>
-      )}
-    </div>
+    <Box {...boxProps}>
+      <div style={{ overflow: 'hidden' }}>
+        <Slide
+          in={selectedGroup == null}
+          direction="right"
+          appear={false}
+          mountOnEnter
+          unmountOnExit
+        >
+          <div>
+            <FilterHeader>
+              <Typography variant="h6">
+                {Object.keys(selectedFilters).length > 0
+                  ? t('Filter ({{count}})', {
+                      count: Object.keys(selectedFilters).length,
+                    })
+                  : t('Filter')}
+              </Typography>
+              <LinkButton
+                color="primary"
+                style={{ marginInlineStart: theme.spacing(-1) }}
+                disabled={Object.keys(selectedFilters).length == 0}
+                onClick={() => alert('TODO')}
+              >
+                {t('Save')}
+              </LinkButton>
+              <LinkButton
+                color="primary"
+                style={{ marginInlineStart: theme.spacing(2) }}
+                disabled={Object.keys(selectedFilters).length == 0}
+                onClick={() => setSelectedFilters({})}
+              >
+                {t('Clear All')}
+              </LinkButton>
+            </FilterHeader>
+            <FilterList dense>
+              {error && (
+                <ListItem data-testid="ErrorState">
+                  <ListItemText
+                    primary={error.toString()}
+                    primaryTypographyProps={{
+                      variant: 'subtitle1',
+                      color: 'error',
+                    }}
+                  />
+                </ListItem>
+              )}
+              {loading ? (
+                <ListItem data-testid="LoadingState">
+                  <CircularProgress />
+                </ListItem>
+              ) : data?.contactFilters?.length == 0 ? (
+                <ListItem data-testid="NoFiltersState">
+                  <ListItemText
+                    primary={t('No Contact Filters Found')}
+                    primaryTypographyProps={{ variant: 'subtitle1' }}
+                  />
+                </ListItem>
+              ) : (
+                <>
+                  {data?.contactFilters?.map((group: ContactFilterGroup) => (
```

Comment:
TS can infer this better than manually updating it since `data` has a type
```suggestion
                  {data?.contactFilters?.map((group) => (
```

Reply:
removing the type there causes a different not assignable type error on the calls to `isGroupVisible(group)` and `showGroup(group)`

Reply:
this did get addressed once I setup the `FilterGroup` alias of the GraphQL type